ROADMAP.md

```diff
@@ -0,0 +1,155 @@
+
+#### 1. [WebTransport](https://github.com/libp2p/js-libp2p-webtransport/issues/1)
+Implementation of WebTransport in js-libp2p. Allows for interoperability with go-libp2p.
+#### 2. [WebRTC: Browser to Server](https://github.com/little-bear-labs/js-libp2p-webrtc/pull/4)
+Add support for WebRTC transport in js-libp2p, enabling browser connectivity with servers. This will cover the browsers that don't support WebTransport (most notable is iOS Safari). This is getting close to finalized.
+#### 3. WebRTC: Browser to Browser
+Even though this use case is made possible by [webrtc-star](https://github.com/libp2p/js-libp2p-webrtc-star) and [webrtc-direct](https://github.com/libp2p/js-libp2p-webrtc-direct) currently, they are a less than ideal solutions. Both libraries have shortcomings, aren't implemented in other languages, and don't employ newer libp2p advancements such as Circuit Relay v2, DCUtR, and authentication via Noise. Therefore, we want to support WebRTC Browser to Browser as a first class transport in js-libp2p and deprecate the previous libraries. A follow up to A.2 where we will begin the work to specify the semantics of browser to browser connectivity and then implement it in js-libp2p.
+
+### B. 🥊 Decentralized Hole Punching
+<!--- TODO: Link to GitHub Epic -->
+**Why**:  P2P networks can have a combination of both public and private nodes. While private nodes can dial nodes on the public Internet, they are unreachable from the outside as they are behind a NAT or a firewall. We need a mechanism to dial them. A [previous DHT crawl found that almost 63%](https://github.com/libp2p/specs/blob/master/ROADMAP.md#-hole-punching-on-tcp-and-quic) of the network was undialable.  This project aims to implement Decentralized Hole Punching in js-libp2p and bring it to parity with the Go and Rust implementations.
+
+**Goal**:
+#### 1. [AutoNat](https://github.com/libp2p/js-libp2p/issues/1005)
+Determine whether a node is public or private (located behind a firewall or a NAT.) This is a dependency for enabling the DHT in server mode by default for projects like js-ipfs.
+#### 2. [Circuit Relay v2](https://github.com/libp2p/js-libp2p/issues/1029)
+Connect to, request reservations, and establish a secure relay connection through discovered public relay node.
+#### 3. Add QUIC Transport
+Hole punching is [more reliable with UDP](https://www.notion.so/pl-strflt/NAT-Hole-punching-Success-Rate-2022-09-29-Data-Analysis-8e72705ca3cc49ab983bc5e8792e3e98#5b76991da8d24736abd486aa93e85a97) (therefore QUIC) than TCP. This requires adding support for QUIC in js-libp2p. There is some work [being done here](https://github.com/nodejs/node/pull/44325) to add support in node.js which we depend on.
+#### 4. Hole Punching
+Use [DCUtR](https://github.com/libp2p/specs/blob/master/relay/DCUtR.md) to synchronize hole punching
+
+### C. 🔮 Future-proof Observability
```

Comment:
Generally seems like something we want to solve/do across libp2p implementations.

We don't have to, but it would be great to see about designing/doing this in a way that makes it possible for multiple implementations to follow suit.

The "benchmarking with remote runners" effort that go-libp2p is easy to push on is related hee.

Reply:
Yes, it's moved to the test-plans milestone now. Updated here.